plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeDynamicFiltering.java

```diff
@@ -0,0 +1,129 @@
+ */
+package io.trino.plugin.deltalake;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.trino.Session;
+import io.trino.execution.DynamicFilterConfig;
+import io.trino.execution.QueryStats;
+import io.trino.operator.OperatorStats;
+import io.trino.plugin.deltalake.util.DockerizedMinioDataLake;
+import io.trino.spi.QueryId;
+import io.trino.sql.planner.OptimizerConfig;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.MaterializedResult;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.ResultWithQueryId;
+import org.testng.annotations.Test;
+
+import static com.google.common.base.Verify.verify;
+import static io.airlift.testing.Assertions.assertEqualsIgnoreOrder;
+import static io.airlift.testing.Assertions.assertGreaterThan;
+import static io.trino.SystemSessionProperties.ENABLE_DYNAMIC_FILTERING;
+import static io.trino.plugin.deltalake.DeltaLakeDockerizedMinioDataLake.createDockerizedMinioDataLakeForDeltaLake;
+import static io.trino.plugin.deltalake.DeltaLakeQueryRunner.DELTA_CATALOG;
+import static io.trino.sql.planner.OptimizerConfig.JoinDistributionType.BROADCAST;
+import static io.trino.sql.planner.OptimizerConfig.JoinDistributionType.PARTITIONED;
+import static io.trino.tpch.TpchTable.LINE_ITEM;
+import static io.trino.tpch.TpchTable.ORDERS;
+import static java.lang.String.format;
+
+public class TestDeltaLakeDynamicFiltering
+        extends AbstractTestQueryFramework
+{
+    private static final String BUCKET_NAME = "delta-lake-test-dynamic-filtering";
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        verify(new DynamicFilterConfig().isEnableDynamicFiltering(), "this class assumes dynamic filtering is enabled by default");
+        DockerizedMinioDataLake dockerizedMinioDataLake = closeAfterClass(createDockerizedMinioDataLakeForDeltaLake(BUCKET_NAME));
+        QueryRunner queryRunner = DeltaLakeQueryRunner.createS3DeltaLakeQueryRunner(
+                DELTA_CATALOG,
+                "default",
+                // Slowing down the query ensures the dynamic filter has enough time to populate.
+                ImmutableMap.of("delta.max-splits-per-second", "3"),
+                dockerizedMinioDataLake.getMinioAddress(),
+                dockerizedMinioDataLake.getTestingHadoop());
+
+        ImmutableList.of(LINE_ITEM, ORDERS).forEach(table -> {
+            String tableName = table.getTableName();
+            dockerizedMinioDataLake.copyResources("io/trino/plugin/deltalake/testing/resources/databricks/" + tableName, tableName);
+            queryRunner.execute(format("CREATE TABLE %s.%s.%s (dummy int) WITH (location = 's3://%s/%3$s')",
+                    DELTA_CATALOG,
+                    "default",
+                    tableName,
+                    BUCKET_NAME));
+        });
+        return queryRunner;
+    }
+
+    @Test(timeOut = 60_000)
```

Comment:
This can be parametrised as 
`@Test(timeOut = 60_000, dataProvider = "joinDistributionTypes")`

```
@DataProvider
    public Object[][] joinDistributionTypes()
    {
        return Stream.of(JoinDistributionType.values())
                .collect(toDataProvider());
    }
```